analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -567,5 +567,31 @@ public void Main<T>() where T : new()
             validator.ValidateContainsOperation(OperationKind.TypeParameterObjectCreation);
             validator.ValidateTag("Value", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
         }
+
+        [TestMethod]
+        public void Literal_Null_SetsNull_CS()
+        {
+            const string code = @"
+Tag(""Before"", arg);
+arg = null;
+Tag(""After"", arg);";
+            var validator = SETestContext.CreateCS(code, ", object arg").Validator;
+            validator.ValidateContainsOperation(OperationKind.Literal);
+            validator.ValidateTag("Before", x => x.Should().BeNull());
```

Comment:
Can we have a Tag, where "Before" is not null, but another constraint, and "After" has the constraint from before and the  `ObjectConstraint.Null`? Maybe `BoolConstraint.True` or something.

Reply:
Such situation cannot happen. If something is known to be `Null`, than it cannot be anything else at the same time.

Reply:
This is true for "Null" but not for "NotNull". Resolving this in favor of:
https://github.com/SonarSource/sonar-dotnet/pull/5884/files#r928601061